internal/k8s/gvrautodetect.go

```diff
@@ -0,0 +1,61 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	projectcontour "github.com/projectcontour/contour/apis/projectcontour/v1"
+
+	ingressroutev1 "github.com/projectcontour/contour/apis/contour/v1beta1"
+
+	serviceapis "sigs.k8s.io/service-apis/api/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+type GVRMap map[schema.GroupVersionResource]cache.SharedIndexInformer
+type GVRInformer struct {
+	Lookup GVRMap
```

Comment:
Suggest that this type will be easier to use with a method, maybe `InformerForResource(*schema.GroupVersionResource)`

Reply:
The aim here is for this type to grow a method that will take the output of the client-go discovery bits (basically a list of available GVRs), and filter this map so it only contains GVRs the apiserver knows about. I don't see it needing to have individual lookups in a public way for now.